aws/resource_aws_eks_cluster.go

```diff
@@ -432,6 +440,7 @@ func expandEksVpcConfigRequest(l []interface{}) *eks.VpcConfigRequest {
 		EndpointPublicAccess:  aws.Bool(m["endpoint_public_access"].(bool)),
 		SecurityGroupIds:      expandStringSet(m["security_group_ids"].(*schema.Set)),
 		SubnetIds:             expandStringSet(m["subnet_ids"].(*schema.Set)),
+		PublicAccessCidrs:     expandStringSet(m["public_access_cidrs"].(*schema.Set)),
```

Comment:
It looks like the EKS API is sensitive to providing an empty list, so we'll first need to check the length then only send the parameter when necessary:

```
--- FAIL: TestAccAWSEksCluster_basic (12.15s)
    testing.go:640: Step 0 error: errors during apply:
        
        Error: error creating EKS Cluster (tf-acc-test-rpe2e): InvalidParameterException: The publicAccessCidrs list should not be empty.
```

e.g.

```go
	vpcConfigRequest := &eks.VpcConfigRequest{
		EndpointPrivateAccess: aws.Bool(m["endpoint_private_access"].(bool)),
		EndpointPublicAccess:  aws.Bool(m["endpoint_public_access"].(bool)),
		SecurityGroupIds:      expandStringSet(m["security_group_ids"].(*schema.Set)),
		SubnetIds:             expandStringSet(m["subnet_ids"].(*schema.Set)),
	}

	if v, ok := m["public_access_cidrs"].(*schema.Set); ok && v.Len() > 0 {
		vpcConfigRequest.PublicAccessCidrs = expandStringSet(v)
	}

	return vpcConfigRequest
```